jib-cli/src/main/java/com/google/cloud/tools/jib/cli/JibCli.java

```diff
@@ -67,6 +81,49 @@ static void logTerminatingException(
             + "\u001B[0m");
   }
 
+  static Future<Optional<String>> newUpdateChecker(
+      GlobalConfig globalConfig, Verbosity verbosity, Consumer<LogEvent> log) {
+    if (!verbosity.atLeast(Verbosity.info) || globalConfig.isDisableUpdateCheck()) {
+      return Futures.immediateFuture(Optional.empty());
+    }
+    ExecutorService executorService = Executors.newSingleThreadExecutor();
+    try {
+      return UpdateChecker.checkForUpdate(
+          executorService, VERSION_URL, VersionInfo.TOOL_NAME, TOOL_VERSION, log);
+    } finally {
+      executorService.shutdown();
+    }
+  }
+
+  @VisibleForTesting
+  static void finishUpdateChecker(
+      ConsoleLogger logger, Future<Optional<String>> updateCheckFuture) {
+    UpdateChecker.finishUpdateCheck(updateCheckFuture)
+        .ifPresent(
+            latestVersion -> {
+              logger.log(
+                  LogEvent.Level.LIFECYCLE,
+                  "\n\u001B[33m"
+                      + "A new version of "
+                      + VersionInfo.TOOL_NAME
+                      + " ("
+                      + latestVersion
+                      + ") is available (currently using "
+                      + TOOL_VERSION
+                      + "). Download the latest jib-cli version from "
+                      + ProjectInfo.GITHUB_URL
+                      + "/releases/tag/v"
+                      + latestVersion
+                      + "-cli"
+                      + "\n"
+                      + ProjectInfo.GITHUB_URL
+                      + "/blob/master/jib-cli/CHANGELOG.md\u001B[0m"
```

Comment:
I think we should have `\n` after this. Maybe `\n\n` is better?

It's a bit hard to parse this and check if this is correctly formed. Maybe `String.format` can improve readability? And I think we can just embed "Jib CLI" without `VersionInfo.TOOL_NAME`.

```java
String message = "A new version of Jib CLI (%d) is available (currently using %d). Download the latest Jib CLI from %s/release/tag/v%d-cli\n"
    + "%s/blob/master/jib-cli/CHANGELOG.md\u001B[0m\n\n"
    + ...
```